06.bowling_object/game.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+require_relative 'shot'
+require_relative 'frame'
+
+class Game
+  def initialize(mark)
+    @mark = mark
+    make_shots
+    @frames = []
+    @count_frames = []
+  end
+
+  def make_shots
+    marks = @mark.split(',')
+
+    @shots = marks.map do |mark|
+      Shot.new(mark).score
+    end
+  end
+
+  def make_frames
+    @shots.each_with_index do |shot, index|
+      count_shot(shot, index)
+    end
+    @frames.map(&:make_frame)
+  end
+
+  def count_shot(shot, index)
+    @count_frames << shot
+    if @frames.size < 9
+      regular_frame(shot)
+    else
+      last_frame(index)
+    end
+  end
+
+  def regular_frame(shot)
+    if @count_frames.size == 2
+      @frames << Frame.new(@count_frames[0], @count_frames[1])
+      @count_frames.clear
+    elsif shot == 10
+      @frames << Frame.new(shot, nil)
+      @count_frames.clear
+    end
+  end
+
+  def last_frame(index)
+    return unless @frames.size == 9
```

Comment:
`unless` は条件式を理解してそれを反転する、という2ステップ必要になるので、個人的には余程簡単な場合以外は `if` を使うことが多いです。

```suggestion
    return if @frames.size != 9
```

ただ、個人の好みの問題もあるので、どちらでもかまいません。

Reply:
ありがとうございます🙏